flow/include/flow/Trace.h

```diff
@@ -278,12 +278,7 @@ struct SWIFT_CXX_IMPORT_OWNED BaseTraceEvent {
 	template <class T>
 	typename std::enable_if<Traceable<T>::value && !std::is_enum_v<T>, BaseTraceEvent&>::type detail(std::string&& key,
 	                                                                                                 const T& value) {
-		if (enabled && init()) {
-			auto s = Traceable<T>::toString(value);
-			addMetric(key.c_str(), value, s);
-			return detailImpl(std::move(key), std::move(s), false);
```

Comment:
I suspect this more efficient. For the other one taking `const char*` as input, we constructing the string `std::string(key)`. So this implementation avoids additional string construction.

Reply:
Yes. The other way seems better.

Reply:
Agreed. Another option is to make `detail(const char* key, const T& value)` call `detail(std::string&& key, const T& value)` and the logic lives in the latter. But the challenge with that is that I will have to unnecessarily create the string in cases where `(enabled && init())` is false, which could be inefficient.

So I will drop this PR.

Reply:
I agree with the comments above. IMHO, if we want to improve the performance, maybe not use `std::string` but `std::string_view` since the keys, in general, are static text.

Reply:
I am not sure about the performance though. I feel the compiler would optimize those constructions, but it may not cause performance improvement.

Reply:
Performance improvement was not the goal of this PR, just reducing duplicate code. Since it may cause perf regression, I will drop this PR.

Reply:
Well, if the motivation is to reduce duplicate code, maybe we could do that in the other way, i.e. let the `const char*` function call the `std::string` function.

Reply:
@xis19 see https://github.com/apple/foundationdb/pull/11504#discussion_r1674998654.